### PR TITLE
249 fix/delete posts by custom taxonomy test

### LINKED ISSUE #249: Delete Posts by Custom Taxonomy - Add tests
## Tests
- [x] Create helper function to create posts in a custom taxonomy and term
- [x] Create helper function to retrieve posts in a custom taxonomy and term
- [x] Add tests to test deleting posts from a single taxonomy term (default post type)
- [x] Add tests to test deleting posts from two taxonomy terms (default post type)
- [x] Add tests to test deleting posts from a single taxonomy term (custom post type)
- [x] Add tests to test deleting posts from two taxonomy terms (custom post type)
- [x] Add tests to test deleting posts permanently ( at least 2 test cases)
- [x] Add tests to test deleting posts that are older than x days (use at least two terms for this)
- [x] Add tests to test deleting posts that are posted within the last x days (use  at least two terms for this)
- [x] Create more than 100 posts and try to delete them in batches. Test at least 2 batches.
- [ ] Test that scheduling works by checking if a cron job is scheduled.

Once this task is done, the helper functions should be moved to the tests library repo.

While sending this task for code review create a new issue for moving the helper functions to the the test library repo

### PR DESCRIPTION
**Pull request by rajanit2000**
Fix #249

### CODE REVIEW

File: tests/wp-unit/include/Core/Posts/Modules/DeletePostsByTaxonomyModuleTest.php [hunk 1]
```diff
@@ -98,5 +98,483 @@ public function test_deleting_posts_from_built_in_taxonomy_terms_in_a_custom_pos
...
+		// Assert that category has no post.
+		$posts = $this->get_posts_by_custom_taxonomy( "custom", "Custom Term" );
+		$this->assertEquals( 50, count( $posts ) );
+
+	}
+
+	/**
+	 * Helper function to get posts by custom taxonomy
+	 */
+	public function get_posts_by_custom_taxonomy( $taxonomy = "custom", $term = "Custom Term", $post_type = "post" ) {
```
Comment (sudar) [inline tests/wp-unit/include/Core/Posts/Modules/DeletePostsByTaxonomyModuleTest.php]: We already have the `get_posts_by_custom_term ` https://github.com/sudar/wp-plugin-test-tools/blob/master/src/Tests/WPCore/WPCoreUnitTestCase.php#L82 helper function

You can use that. No need to create a new one.

File: tests/wp-unit/include/Core/Posts/Modules/DeletePostsByTaxonomyModuleTest.php [hunk 2]
```diff
@@ -98,5 +98,483 @@ public function test_deleting_posts_from_built_in_taxonomy_terms_in_a_custom_pos
 		$posts_in_cat1 = $this->get_posts_by_category( $cat1, 'custom' );
 
 		$this->assertEquals( 0, count( $posts_in_cat1 ) );
+
+	}
+
+	public function test_that_trash_posts_from_single_taxonomy_term() {
```
Comment (sudar) [inline tests/wp-unit/include/Core/Posts/Modules/DeletePostsByTaxonomyModuleTest.php]: Add phpdoc block comments to all methods.

**Comment by sudar** [comment]
@mariadanieldeepak 

Review these test cases and give your feedback to @rajanit2000

**Comment by rajanit2000** [comment]
@mariadanieldeepak its ready to review

File: tests/wp-unit/include/Core/Posts/Modules/DeletePostsByTaxonomyModuleTest.php [hunk 3]
```diff
@@ -26,6 +26,9 @@ public function setUp() {
 		$this->module = new DeletePostsByTaxonomyModule();
 	}
 
+	/**
+	 * Test deleting posts from a single taxonomy term default post type.
+	 */
 	public function test_deleting_posts_from_built_in_taxonomy_terms() {
```
Comment (mariadanieldeepak) [inline tests/wp-unit/include/Core/Posts/Modules/DeletePostsByTaxonomyModuleTest.php]: @rajanit2000 - The function name says that posts are `deleted`. But looking at the `force_delete` filter, you're only `trashing` and **not** `deleting`. Kindly **change** the function name respectively.

Check the following line - https://github.com/sudar/bulk-delete/pull/382/files#diff-de80f6e764d3cf5960c34228622a60d4R50

Comment (mariadanieldeepak) [inline tests/wp-unit/include/Core/Posts/Modules/DeletePostsByTaxonomyModuleTest.php]: L#48 `private` filter is not required here, as **we do not have an option in the UI** for users to set this.

Comment (mariadanieldeepak) [inline tests/wp-unit/include/Core/Posts/Modules/DeletePostsByTaxonomyModuleTest.php]: https://github.com/sudar/bulk-delete/pull/382/files#diff-de80f6e764d3cf5960c34228622a60d4R49

`limit_to` filter should be removed.

Comment (mariadanieldeepak) [inline tests/wp-unit/include/Core/Posts/Modules/DeletePostsByTaxonomyModuleTest.php]: https://github.com/sudar/bulk-delete/pull/382/files#diff-de80f6e764d3cf5960c34228622a60d4R47

`restrict` should be removed.

Comment (mariadanieldeepak) [inline tests/wp-unit/include/Core/Posts/Modules/DeletePostsByTaxonomyModuleTest.php]: 'date_op' should be removed. See why [here](https://github.com/sudar/bulk-delete/blob/dev/6.0.0/include/util/query.php#L44).

https://github.com/sudar/bulk-delete/pull/382/files#diff-de80f6e764d3cf5960c34228622a60d4R51

Comment (mariadanieldeepak) [inline tests/wp-unit/include/Core/Posts/Modules/DeletePostsByTaxonomyModuleTest.php]: Avoid hard-coding `cat1` in https://github.com/sudar/bulk-delete/pull/382/files#diff-de80f6e764d3cf5960c34228622a60d4R46

Comment (mariadanieldeepak) [inline tests/wp-unit/include/Core/Posts/Modules/DeletePostsByTaxonomyModuleTest.php]: Fix formatting on the following lines.

https://github.com/sudar/bulk-delete/blob/a239562313b008b47c6681fc3ff4c34fb305ed02/tests/wp-unit/include/Core/Posts/Modules/DeletePostsByTaxonomyModuleTest.php#L45-L52

Comment (mariadanieldeepak) [inline tests/wp-unit/include/Core/Posts/Modules/DeletePostsByTaxonomyModuleTest.php]: When writing comments, start the first letter in Caps.

https://github.com/sudar/bulk-delete/blob/a239562313b008b47c6681fc3ff4c34fb305ed02/tests/wp-unit/include/Core/Posts/Modules/DeletePostsByTaxonomyModuleTest.php#L71

Comment (mariadanieldeepak) [inline tests/wp-unit/include/Core/Posts/Modules/DeletePostsByTaxonomyModuleTest.php]: When using the value `custom` in more than one line, use a variable.

https://github.com/sudar/bulk-delete/blob/a239562313b008b47c6681fc3ff4c34fb305ed02/tests/wp-unit/include/Core/Posts/Modules/DeletePostsByTaxonomyModuleTest.php#L77-L78

https://github.com/sudar/bulk-delete/blob/a239562313b008b47c6681fc3ff4c34fb305ed02/tests/wp-unit/include/Core/Posts/Modules/DeletePostsByTaxonomyModuleTest.php#L80

https://github.com/sudar/bulk-delete/blob/a239562313b008b47c6681fc3ff4c34fb305ed02/tests/wp-unit/include/Core/Posts/Modules/DeletePostsByTaxonomyModuleTest.php#L82

https://github.com/sudar/bulk-delete/blob/a239562313b008b47c6681fc3ff4c34fb305ed02/tests/wp-unit/include/Core/Posts/Modules/DeletePostsByTaxonomyModuleTest.php#L104

Comment (mariadanieldeepak) [inline tests/wp-unit/include/Core/Posts/Modules/DeletePostsByTaxonomyModuleTest.php]: Format the following lines as per the [WordPress Coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/php/#indentation).

https://github.com/sudar/bulk-delete/blob/a239562313b008b47c6681fc3ff4c34fb305ed02/tests/wp-unit/include/Core/Posts/Modules/DeletePostsByTaxonomyModuleTest.php#L88-L96

Comment (mariadanieldeepak) [inline tests/wp-unit/include/Core/Posts/Modules/DeletePostsByTaxonomyModuleTest.php]: Avoid function naming inconsistency across the different test cases. Either use `test_deleting` or `test_that_posts_are_deleted_...`

https://github.com/sudar/bulk-delete/blob/a239562313b008b47c6681fc3ff4c34fb305ed02/tests/wp-unit/include/Core/Posts/Modules/DeletePostsByTaxonomyModuleTest.php#L32

https://github.com/sudar/bulk-delete/blob/a239562313b008b47c6681fc3ff4c34fb305ed02/tests/wp-unit/include/Core/Posts/Modules/DeletePostsByTaxonomyModuleTest.php#L113

Comment (mariadanieldeepak) [inline tests/wp-unit/include/Core/Posts/Modules/DeletePostsByTaxonomyModuleTest.php]: @rajanit2000,
 
The above review comments are applicable to all test cases. Kindly correct/fix test cases and send them back for review.

**Comment by rajanit2000** [comment]
@mariadanieldeepak I have updated the review comments. please review it again

**Comment by mariadanieldeepak** [comment]
@rajanit2000 - The PR doesn't any description. Kindly refer the [WIKI](https://github.com/sudar/bulk-wp-site/wiki/Workflow#github-pull-request-submission-workflow) on how to write PR descriptions.

**Comment by rajanit2000** [comment]
Its ready for review, Please review it again

**Comment by mariadanieldeepak** [comment]
@rajanit2000,

I would suggest you to run `phpcs` on your files. Code Sniffer as of now shows 110 errors (and 44 warnings). Most of them are repeated from our 1st review.

I would love you to tackle all of 'em.

**Comment by rajanit2000** [comment]
@sudar its ready to review

**Comment by rajanit2000** [comment]
@mariadanieldeepak i have done the phpcs issues, Now its ready to review

**Comment by mariadanieldeepak** [comment]
@rajanit2000 Thank you, will review it. Also, make sure you merge the latest `dev/6.0.0` each time you send the code for the review.

**Comment by rajanit2000** [comment]
@mariadanieldeepak i have update the latest `dev/6.0.0`  branch. Now its ready to review

**Comment by mariadanieldeepak** [comment]
@rajanit2000 

Could you merge the latest `dev/6.0.0`?

**Comment by rajanit2000** [comment]
image
@sudar I have an issue when running the PHP unit globally. Its working fine when running individually

**Comment by sudar** [comment]
@rajanit2000 

The above issue that you reported was caused because of incorrect registration of post type and taxonomy.

I have fixed it now and the entire test suite is running.

Fix the remaining issues, do a self-review and then send the code fore review again.

**Comment by rajanit2000** [comment]
@sudar its ready to review

File: include/Core/Posts/Modules/DeletePostsByTaxonomyModule.php [hunk 4]
```diff
@@ -12,6 +12,9 @@
  * @since 6.0.0
  */
 class DeletePostsByTaxonomyModule extends PostsModule {
+	/**
```
Comment (sudar) [inline include/Core/Posts/Modules/DeletePostsByTaxonomyModule.php]: @rajanit2000 

Why is this file changed?

I am seeing lot of whitespace changes, so I am not sure what changes you made to this file.

File: include/Core/Posts/PostsModule.php [hunk 5]
```diff
@@ -111,21 +125,170 @@ protected function do_delete( $options ) {
...
 	 * Render the "private post" setting fields.
 	 */
 	protected function render_private_post_settings() {
-		if( $this->are_private_posts_present() ){
+		if ( $this->are_private_posts_present() ) {
 			bd_render_private_post_settings( $this->field_slug );
 		}
 	}
 
+	/**
```
Comment (sudar) [inline include/Core/Posts/PostsModule.php]: All these methods are moved to the `Renderer` method. Why are they added back to `PostsModules`?

File: include/Core/Posts/Modules/DeletePostsByTaxonomyModule.php (hunk 4 above)
Comment (rajanit2000) [reply include/Core/Posts/Modules/DeletePostsByTaxonomyModule.php]: Its update when run phpcbf command, How can i revert it?
